pfcpiface/upf.go

```diff
@@ -68,6 +73,30 @@ func (u *upf) sim(method string) {
 }
 
 func (u *upf) setUpfInfo(conf *Conf) {
+	u.n4SrcIP = net.ParseIP("0.0.0.0")
+	u.nodeIP = net.ParseIP("0.0.0.0")
+
+	if conf.CPIface.SrcIP == "" {
+		if conf.CPIface.DestIP != "" {
+			log.Println("Dest address ", conf.CPIface.DestIP)
+			u.n4SrcIP = getLocalIP(conf.CPIface.DestIP)
+			log.Println("SPGWU/UPF address IP: ", u.n4SrcIP.String())
+		}
+	} else {
+		addrs, errin := net.LookupHost(conf.CPIface.SrcIP)
+		if errin == nil {
+			u.n4SrcIP = net.ParseIP(addrs[0])
+		}
+	}
+
+	if conf.CPIface.FQDNHost != "" {
+		ips, err := net.LookupHost(conf.CPIface.FQDNHost)
+		if err == nil {
+			u.nodeIP = net.ParseIP(ips[0])
+		}
+	}
+	log.Println("UPF Node IP : ", u.nodeIP.String())
+	log.Println("UPF Local IP : ", u.n4SrcIP.String())
```

Comment:
What happens when - 

1. FQDNHost is not present - it should use `u.n4SrcIP`
2. Neither SrcIP nor DestIP is present - We are simply listening on 0.0.0.0 and when connection is initiated from control plane we should fill up `u.n4SrcIP` and `u.nodeIP`